plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MysqlTableProperties.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mysql;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.trino.plugin.jdbc.TablePropertiesProvider;
+import io.trino.spi.session.PropertyMetadata;
+import io.trino.spi.type.ArrayType;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class MysqlTableProperties
+        implements TablePropertiesProvider
+{
+    private final List<PropertyMetadata<?>> tableProperties;
+
+    public static final String PRIMARY_KEY_PROPERTY = "primary_key";
+
+    @Inject
+    public MysqlTableProperties()
+    {
+        tableProperties = ImmutableList.of(
```

Comment:
Use `ImmutableList.Builder` so we can reduce diff when adding a new entry to the bottom.